citra-qt: fixes to per game settings

vitor-k opened the pull request:
Removes some unused functions and fixes some issues related to the per-game settings.

fixes #6238

fixes the settings that were written as QMetaTypes, like `@Variant(\0\0\0\x85\0\0)`, which would make them unreadable by humans.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/citra-emu/citra/6298)
<!-- Reviewable:end -->

GPUCode commented on src/citra_qt/configuration/config.cpp:
```diff
@@ -181,6 +181,15 @@ void Config::WriteBasicSetting(const Settings::Setting<std::string>& setting) {
+    const QString name = QString::fromStdString(setting.GetLabel());
+    const u16& value = setting.GetValue();
+    qt_config->setValue(name + QStringLiteral("/default"), value == setting.GetDefault());
+    qt_config->setValue(name, static_cast<unsigned int>(value));
```
```suggestion
    qt_config->setValue(name, static_cast<u32>(value));
```

GPUCode commented on src/citra_qt/configuration/config.cpp:
```diff
@@ -223,6 +232,20 @@ void Config::WriteGlobalSetting(const Settings::SwitchableSetting<std::string>&
+    }
+    if (global || !setting.UsingGlobal()) {
+        qt_config->setValue(name + QStringLiteral("/default"), value == setting.GetDefault());
+        qt_config->setValue(name, static_cast<unsigned int>(value));
```
ditto

GPUCode commented on src/common/settings.h:
```diff
@@ -201,6 +201,16 @@ class Setting {
+     *
+     * @param val The desired value
+     */
+    virtual void SetDefault(const Type& val) {
```
This doesn't need to be virtual

vitor-k commented:
Addressed review comments and changed the post processing name setting to be split into one for the regular post processing and one for the anaglyph (as they already select from two different pools of options). With this change the selected shader will be remembered when switching from anaglyph and non-anaglyph 3d options.
Interlaced/reverse interlaced is left out, but the UI was reworked so that the selected shader can't be changed when using it.
This reflects the code behavior better (you can't use a post processing shader at the same time as the interlaced 3d options, it'd overwrite it).
